### PR TITLE
Allocation Command

### PR DESCRIPTION
**Pull request by northern-64bit**
# Description

Adds the allocation command to the portfolio context. After loading a portfolio and using the command, a table (or pie chart if choosen) of the allocation gets displayed. The allocation command takes etf holdings into consideration and adds them to the table or directly to an existing stock position. This is also preparing for allocation handling on the standard portfolio report.

# How has this been tested?

It has been tested by using it on different portfolios with different etfs and using all parameters. It didn't break or output any errors.

# Checklist:

- [x] Update [our Hugo documentation](https://gamestonkterminal.github.io/GamestonkTerminal/) following [these guidelines](https://github.com/GamestonkTerminal/GamestonkTerminal/tree/main/website).
- [x] Make sure you are following our [CONTRIBUTING guidelines](https://github.com/GamestonkTerminal/GamestonkTerminal/blob/main/CONTRIBUTING.md).
- [ ] Update our tests following [these guidelines](https://github.com/GamestonkTerminal/GamestonkTerminal/tree/main/tests).
I don't know how to do this
- [ ] If a feature was added make sure to add it to the corresponding [scripts file](https://github.com/GamestonkTerminal/GamestonkTerminal/tree/main/scripts).
I don't know what to do here, help? Isn't this some kind of code test?

# Others
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My code passes all the checks pylint, flake8, black, ... To speed up development you should run `pre-commit install`.
- [x] New and existing unit tests pass locally with my changes. You can test this locally using `pytest tests/...`.

### CODE REVIEW

File: gamestonk_terminal/portfolio/portfolio_view.py [hunk 1]
```diff
@@ -251,6 +251,39 @@ def plot_ef(
     return ImageReader(imgdata)
 
 
+def al(data: pd.DataFrame, graph: bool):
```
Comment (colin99d) [inline gamestonk_terminal/portfolio/portfolio_view.py:254]: This function name is a little ambiguous. I would rename to something like display_allocation

Comment (northern-64bit) [reply gamestonk_terminal/portfolio/portfolio_view.py]: Fixed :)

**Comment by colin99d** [comment]
Screen Shot 2022-01-18 at 2 37 56 PM
Looks really good so far. Is there a way to make the pie chart prettier when the slices are really small?

**Comment by northern-64bit** [comment]
> Looks really good so far. Is there a way to make the pie chart prettier when the slices are really small?

bild
Something similar to this may be a possible solution, but Idk if it's a good one

**Comment by colin99d** [comment]
> ! Something similar to this (but with the portfolio) may be a possible solution and then show the small holdings on the side thing, but Idk if it's a good one

Is there a way to have the key on the side instead of in the graph?

**Comment by DidierRLopes** [comment]
> > Looks really good so far. Is there a way to make the pie chart prettier when the slices are really small?
> 
> bild Something similar to this (but with the portfolio) may be a possible solution and then show the small holdings on the side thing, but Idk if it's a good one

we can probably use "Others" when slice is too small and then display in print all holdings.

Also, you can use parameter `explode` so the slices come off a bit if you want, not sure if it is helpful in this case though

**Comment by northern-64bit** [comment]
> > ! Something similar to this (but with the portfolio) may be a possible solution and then show the small holdings on the side thing, but Idk if it's a good one
> 
> Is there a way to have the key on the side instead of in the graph?

bild
Having the keys in the labels is an option like this but with better formatting

> we can probably use "Others" when slice is too small and then display in print all holdings.

This is also an option, but it could maybe be solved by the approach in the picture above maybe.

**Review by jmaslek — APPROVED** [review]

**Pull request merged**